tests/status/test_legacy_tracker.py

```diff
@@ -1,38 +1,75 @@
-from ert_shared.feature_toggling import FeatureToggling
+import os
+import re
+import shutil
+import threading
+from argparse import ArgumentParser
+
+import pytest
+from res.enkf.enkf_main import EnKFMain
+from res.enkf.res_config import ResConfig
+
+import ert_shared.status.entity.state as state
+from ert_shared import ERT
+from ert_shared.cli import (
+    ENSEMBLE_SMOOTHER_MODE,
+    ENSEMBLE_EXPERIMENT_MODE,
+)
+from ert_shared.cli.model_factory import create_model
+from ert_shared.cli.notifier import ErtCliNotifier
 from ert_shared.ensemble_evaluator.config import EvaluatorServerConfig
-from ert_shared.ensemble_evaluator.entity.snapshot import PartialSnapshot, Snapshot
+from ert_shared.feature_toggling import FeatureToggling
+from ert_shared.main import ert_parser
 from ert_shared.status.entity.event import (
     EndEvent,
     FullSnapshotEvent,
     SnapshotUpdateEvent,
 )
-from ert_shared.status.entity.state import (
-    JOB_STATE_FINISHED,
-    REALIZATION_STATE_FINISHED,
-)
 from ert_shared.status.tracker.factory import create_tracker
-import threading
-from ert_shared.cli.model_factory import create_model
-from ert_shared.cli.notifier import ErtCliNotifier
-import shutil
-import os
-from argparse import ArgumentParser
-from ert_shared.main import ert_parser
-from ert_shared.cli import (
-    ENSEMBLE_SMOOTHER_MODE,
-    ENSEMBLE_EXPERIMENT_MODE,
-    ITERATIVE_ENSEMBLE_SMOOTHER_MODE,
-    ES_MDA_MODE,
-)
-from res.enkf.enkf_main import EnKFMain
-from res.enkf.res_config import ResConfig
-from ert_shared import ERT
-import pytest
+
+
+def check_expression(original, path_expressions, expected, msg_start, path=""):
+    assert isinstance(original, dict), f"{msg_start}data is not a dict"
+    pattern = path_expressions[0]
+    path_expressions = path_expressions[1:]
+    matched = False
+    for key, value in original.items():
+        current_path = f"{path}/{key}"
+        if re.match(pattern, key):
+            matched = True
+            if path_expressions:
+                assert isinstance(value, dict), f"{msg_start}{current_path} not a dict"
+                check_expression(
+                    value, path_expressions, expected, msg_start, current_path
+                )
+            else:
+                assert (
+                    expected == value
+                ), f"{msg_start}{current_path} value ({original[key]}) is not equal to ({value})"
+    assert matched, f"{msg_start}{pattern} did not match anything in {path}"
```

Comment:
Isn't this a home grown https://pypi.org/project/jsonpath-ng/ though it only works on python dicts?

This is probably going to be a thing if we move forward with narratives, so why not just implement it right away? We've long decided that these snapshots can be serialized.